src/subscription_manager/utils.py

```diff
@@ -377,12 +377,12 @@ def set_filter_string(self, filter_string):
             '?': '.',
         }
 
-        expression = ur"""
+        expression = u"""
             ((?:                # A captured, non-capture group :)
-                [^*?\\]*        # Character literals and other uninteresting junk (greedy)
-                (?:\\.?)*       # Anything escaped with a backslash, or just a trailing backslash
+                [^*?\\\\]*        # Character literals and other uninteresting junk (greedy)
+                (?:\\\\.?)*       # Anything escaped with a backslash, or just a trailing backslash
             )*)                 # Repeat the above sequence 0+ times, greedily
-            ([*?]|\Z)           # Any of our wildcards (* or ?) not preceded by a backslash OR end of input
+            ([*?]|\\Z)           # Any of our wildcards (* or ?) not preceded by a backslash OR end of input
```

Comment:
Were these regexs wrong?  As I understand it, the new regex will match "\Z" instead of end of input.

Reply:
> Were these regexs wrong? As I understand it, the new regex will match "\Z" instead of end of input.

Don't think so. Reason for this change was that raw unicode literals were removed from Python 3... and when running setup.py, we end up importing or otherwise processing `utils.py` (don't recall specifics)... and so I needed to change the literal to a plain unicode literal... This meant double escaping.

You can do this to sanity-check though: take both representations and confirm that they produce equivalent strings:

```python
expression1 = ur"""
            ((?:                # A captured, non-capture group :)
                [^*?\\]*        # Character literals and other uninteresting junk (greedy)
                (?:\\.?)*       # Anything escaped with a backslash, or just a trailing backslash
            )*)                 # Repeat the above sequence 0+ times, greedily
            ([*?]|\Z)           # Any of our wildcards (* or ?) not preceded by a backslash OR end of input
        """

expression2 = u"""
            ((?:                # A captured, non-capture group :)
                [^*?\\\\]*        # Character literals and other uninteresting junk (greedy)
                (?:\\\\.?)*       # Anything escaped with a backslash, or just a trailing backslash
            )*)                 # Repeat the above sequence 0+ times, greedily
            ([*?]|\\Z)           # Any of our wildcards (* or ?) not preceded by a backslash OR end of input
        """

print(expression1 == expression2)
```

outputs `True`